amr-wind/wind_energy/ABLStats.cpp

```diff
@@ -429,9 +440,85 @@ void ABLStats::write_netcdf()
+                "u'v'_sfs", "u'w'_sfs", "v'w'_sfs"};
+            for (int i = 0; i < AMREX_SPACEDIM; i++) {
+                m_pa_sfs->line_average(i, l_vec);
+                auto var = grp.var(var_names[i]);
+                var.put(l_vec.data(), start, count);
+            }
+        }
+
+        {
+            amrex::Vector<std::string> var_names{
+                "u'theta'_sfs", "v'theta'_sfs", "w'theta'_sfs"};
+            for (int i = 0; i < AMREX_SPACEDIM; i++) {
+                m_pa_tsfs->line_average(i, l_vec);
+                auto var = grp.var(var_names[i]);
+                var.put(l_vec.data(), start, count);
+            }
+        }
     }
     ncf.close();
 #endif
 }
 
+void ABLStats::calc_sfs_stress_avgs()
+{
+
+    BL_PROFILE("amr-wind::ABLStats::calc_sfs_stres");
+
+    auto& repo = m_sim.repo();
+
+    auto sfs_stress = repo.create_scratch_field("sfs_stress", 3);
+    auto t_sfs_stress = repo.create_scratch_field("tsfs_stress", 3);
+
+    auto& m_vel = repo.get_field("velocity");
+    auto gradVel = repo.create_scratch_field(9);
+    fvm::gradient(*gradVel, m_vel);
+
+    auto& alphaeff = repo.get_field(pde_impl::mueff_name("temperature"));
+    auto gradT = repo.create_scratch_field(3);
+    fvm::gradient(*gradT, m_temperature);
+
+    const int nlevels = repo.num_active_levels();
+    for (int lev = 0; lev < nlevels; ++lev) {
+        for (amrex::MFIter mfi(m_mueff(lev)); mfi.isValid(); ++mfi) {
+            const auto& bx = mfi.tilebox();
+            const auto& mueff_arr = m_mueff(lev).array(mfi);
+            const auto& alphaeff_arr = alphaeff(lev).array(mfi);
+            const auto& gradVel_arr = (*gradVel)(lev).array(mfi);
+            const auto& gradT_arr = (*gradT)(lev).array(mfi);
+            const auto& sfs_arr = (*sfs_stress)(lev).array(mfi);
+            const auto& t_sfs_arr = (*t_sfs_stress)(lev).array(mfi);
+
+            amrex::ParallelFor(
+                bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                    sfs_arr(i, j, k, 0) =
+                        mueff_arr(i, j, k) *
+                        (gradVel_arr(i, j, k, 1) + gradVel_arr(i, j, k, 4));
+                    sfs_arr(i, j, k, 1) =
+                        mueff_arr(i, j, k) *
+                        (gradVel_arr(i, j, k, 2) + gradVel_arr(i, j, k, 6));
+                    sfs_arr(i, j, k, 2) =
+                        mueff_arr(i, j, k) *
+                        (gradVel_arr(i, j, k, 5) + gradVel_arr(i, j, k, 7));
+
+                    for (int icomp = 0; icomp < AMREX_SPACEDIM; icomp++)
+                        t_sfs_arr(i, j, k, icomp) =
+                            alphaeff_arr(i, j, k) * gradT_arr(i, j, k, icomp);
+                });
+        }
+    }
+
+    m_pa_sfs =
+        new ScratchFieldPlaneAveraging(*sfs_stress, m_sim.time(), m_normal_dir);
+    (*m_pa_sfs)();
+    m_pa_tsfs = new ScratchFieldPlaneAveraging(
+        *t_sfs_stress, m_sim.time(), m_normal_dir);
+    (*m_pa_tsfs)();
+}
```

Comment:
These can simply be 
```suggestion
    ScratchFieldPlaneAveraging pa_sfs(*sfs_stress, m_sim.time(), m_normal_dir);
    ScratchFieldPlaneAveraging pa_tsfs(*t_sfs_stress, m_sim.time(), m_normal_dir);
    pa_sfs();
    pa_tsfs();
    
    // Insert netcdf output here.
}
```

Reply:
Tried addressing in https://github.com/Exawind/amr-wind/pull/290/commits/22cbdfe875f7c910a3fd1369c871a361c06ac0dd